packages/ag-charts-community/src/chart/series/cartesian/util.ts

```diff
@@ -0,0 +1,57 @@
+import type { AgSeriesSegmentation } from 'ag-charts-types';
+
+import type { Scale } from '../../../scale/scale';
+import { BBox } from '../../../scene/bbox';
+import { findRangeExtent } from '../../../util/number';
+import type { ChartAxisDirection } from '../../chartAxisDirection';
+
+export function calculateSegments(
+    segmentation: AgSeriesSegmentation,
+    scales: { [key in ChartAxisDirection]?: Scale<unknown, number> }
```

Comment:
I think you can get away with this

```suggestion
    scales: { [key in (ChartAxisDirection.X | ChartAxisDirection.Y)]: Scale<unknown, number> }
```